protocol/consistencychecks.go

```diff
@@ -245,7 +245,11 @@ func (cc *ConsistencyChecks) updateTBs(requestType int, msg *Response,
 	case RegistrationType:
 		df := msg.DirectoryResponse.(*DirectoryProof)
 		if df.AP.ProofType() == m.ProofOfAbsence {
-			if err := cc.verifyReturnedPromise(df, key); err != nil {
+			tbKey := key
+			if msg.Error == ReqNameExisted {
+				tbKey = df.TB.Value
+			}
```

Comment:
Maybe the service doesn't use a bot though. Maybe I'm just overthinking about this.

> Again, I think this is a case covered above where the user themself needs to be notified.

Yep, I agree with this.